content-moderation/02-video-moderation/02-content-moderation-video-with-human-in-the-loop.ipynb

```diff
@@ -0,0 +1,626 @@
+    "s3.upload_file('../datasets/moderation-video.mp4', data_bucket, s3_key)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Step 2: Moderate video using Rekognition moderation API  <a id=\"step2\"></a>\n",
+    "Call Rekognition Content Moderation API to detect inappropriate information in the video. Rekognition Video moderation API is an asynchronize API that will start a job managed by Rekognition. We will receive a job_id back when we start the job. Then use iteration logic to heart-beat the get_content_moderation API to check job status until the job completes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "startModerationLabelDetection = rekognition.start_content_moderation(\n",
+    "        Video={\n",
+    "            'S3Object': {\n",
+    "                'Bucket': data_bucket,\n",
+    "                'Name': s3_key,\n",
+    "            }\n",
+    "        },\n",
+    "        #MinConfidence=min_confidence,\n",
+    ")\n",
+    "\n",
+    "moderationJobId = startModerationLabelDetection['JobId']\n",
+    "display(\"Job Id: {0}\".format(moderationJobId))\n",
+    "\n",
+    "getContentModeration = rekognition.get_content_moderation(\n",
+    "    JobId=moderationJobId,\n",
+    "    SortBy='TIMESTAMP'\n",
+    ")\n",
+    "\n",
+    "while(getContentModeration['JobStatus'] == 'IN_PROGRESS'):\n",
+    "    time.sleep(5)\n",
+    "    print('.', end='')\n",
+    "\n",
+    "    getContentModeration = rekognition.get_content_moderation(\n",
+    "    JobId=moderationJobId,\n",
+    "    SortBy='TIMESTAMP')\n",
+    "\n",
+    "display(getContentModeration['JobStatus'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "getContentModeration"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As we can see in the Rekognition response, the get_content_moderation API returns a list of labels under the \"ModerationLabels\" field, a list of items each contains:\n",
+    "- Timestamp (millisecond): when inapproperiate information found within the video\n",
+    "- ModerationLabel:\n",
+    "    - Confidence: Confidence score of the label\n",
+    "    - Name: second level moderation category name\n",
+    "    - ParentName: top-level moderation category name\n",
+    "\n",
+    "There are multiple items returned from Rekognition, and hard to review this result by looking at the data directly. In the next step, we will set up A2I so the human reviewers can check these results quickly."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Step 3: Set up A2I for human review <a id=\"step3\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Setup Bucket CORS\n",
```

Comment:
Haven't we done this in lab 01 already? Can we remove to simplify?

Reply:
This code is safe to run twice. The customer may proceed to lab 02 without going through the image lab. So it is needed.